tests/cli/test_report.py

```diff
@@ -64,6 +64,9 @@ def test_report_format():
     # buildtest report --formatfields
     report.print_raw_format_fields()
 
+    # buildtest report --detailed
+    report = Report(configuration=configuration, format_detailed=True)
+
```

Comment:
this is a good test. I think we should update the `__init__` method of `Report` class to also raise an exception if we invoke the following

```python
Report(configuration=configuration,format_detailed=True,format="name,id")
```

Right now we have the exception raised from argparse but there is nothing stop it from happening when invoking the class method.

I think you should have this as a separate method call it `test_report_detailed` where you this negative assertion and also move the code

```python
# buildtest report --detailed
report = Report(configuration=configuration, format_detailed=True)
```

Reply:
@shahzebsiddiqui This is not clear to me, I am trying to fix it since yesterday, I used this code:
`        try:
            report=Report(configuration=configuration, format_detailed=True, format_args="name,id,user")
        except argparse.ArgumentError:
            console.print_exception()
            return True
`
**The result when I run** ` buildtest report --detailed --format name,id`:
(anwar)` anwar@DOE6991925 buildtest % buildtest report --detailed --format name,id
usage: buildtest [options] [COMMANDS] report [-h] [--pager] [--row-count] [-t]
                                             [-n] [-c COUNT] [--filter FILTER]
                                             [--helpfilter] [--filterfields]
                                             [--helpformat] [--formatfields]
                                             [--format FORMAT | -d] [-f | -p]
                                             [-s START] [-e END] [--latest]
                                             [--oldest]
                                             ...
buildtest [options] [COMMANDS] report: error: argument --format: not allowed with argument -d/--detailed
` 
**But when I run** ` buildtest report --detailed `
` 132 │   │   try:                                                           │
│ ❱ 133 │   │   │   report=Report(configuration=configuration, format_detailed │
│   134 │   │   except argparse.ArgumentError:                                 │
│   135 │   │   │   console.print_exception()                                  │
│   136 │   │   │   return True                                                │
│                                                                              │
│ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
│ │           color = None                                                   │ │
│ │   configuration = <buildtest.config.SiteConfiguration object at          │ │
│ │                   0x107b48190>                                           │ │
│ │           count = None                                                   │ │
│ │        detailed = None                                                   │ │
│ │             end = None                                                   │ │
│ │         failure = None                                                   │ │
│ │     filter_args = None                                                   │ │
│ │     format_args = 'name,id,user'                                         │ │
│ │ format_detailed = True                                                   │ │
│ │          latest = None                                                   │ │
│ │          oldest = None                                                   │ │
│ │           pager = None                                                   │ │
│ │          passed = None                                                   │ │
│ │     report_file = None                                                   │ │
│ │            self = <buildtest.cli.report.Report object at 0x109c304c0>    │ │
│ │           start = None                                                   │ │
│ ╰──────────────────────────────────────────────────────────────────────────╯ │
╰──────────────────────────────────────────────────────────────────────────────╯
RecursionError: maximum recursion depth exceeded
`
Any advise?

Reply:
Please fix your message above it's not readable. 

You need to format your message properly

Reply:
updated the text

Reply:
I don't understand what you are doing updating the Report class in __init__ was meant to catch exception as part of regression test but your output above is sharing an invocation to buildtest. Either I am not clear in what I am asking or you have trouble conveying your message. Note you don't need to share the code you want to try just commit it so I can see the changes. 

It's better you just share the screenshot for error and please delete your message.